src/services/expenseServices.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+let expenses = [];
+
+function resetExpenses() {
+  expenses = [];
+};
+
+function getExpenses(userId, categories, from, to) {
+  let filteredExpenses = expenses;
+
+  if (userId) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.userId === +userId
+    );
+  }
+
+  if (categories) {
+    filteredExpenses = filteredExpenses.filter((expense) =>
+      categories.includes(expense.category)
+    );
+  }
+
+  if (from && to) {
+    filteredExpenses = filteredExpenses.filter(expense => {
+      const expenseDate = new Date(expense.spentAt);
+      const fromDate = new Date(from);
+      const toDate = new Date(to);
+
+      return expenseDate < toDate && fromDate <= expenseDate;
+    });
+  }
+
+  return filteredExpenses;
+};
+
+function getExpenseById(id) {
+  const foundExpense = expenses.find((expense) => expense.id === +id);
+
+  return foundExpense;
+};
+
+function createExpense(userId, spentAt, title, amount, category, note) {
+  const newExpense = {
+    id: expenses.length + 1,
```

Comment:
Use a more robust method for generating IDs, like UUID, to avoid potential collisions when creating new expenses.
Or make a separate function for find max id & plus 1 to this value